components/tensorflow-notebook-image/Dockerfile

```diff
@@ -93,71 +88,29 @@ RUN cd /tmp && \
     mv tini /usr/local/bin/tini && \
     chmod +x /usr/local/bin/tini
 
-# Install conda as jovyan user and check the md5 sum provided on the download site
-# After Miniconda v4.5.4 the default Python version is no longer 3.6, but TensorFlow
-# still doesn't support Python 3.7. If we still like to upgrade Miniconda we need
-# to add the line "conda install python==3.6" to RUN command below
-ENV MINICONDA_VERSION 4.5.4
-RUN cd /tmp && \
-    mkdir -p $CONDA_DIR && \
-    wget --quiet https://repo.continuum.io/miniconda/Miniconda3-${MINICONDA_VERSION}-Linux-x86_64.sh && \
-    echo "a946ea1d0c4a642ddf0c3a26a18bb16d *Miniconda3-${MINICONDA_VERSION}-Linux-x86_64.sh" | md5sum -c - && \
-    /bin/bash Miniconda3-${MINICONDA_VERSION}-Linux-x86_64.sh -f -b -p ${CONDA_DIR} && \
-    rm Miniconda3-${MINICONDA_VERSION}-Linux-x86_64.sh && \
-    conda config --system --prepend channels conda-forge && \
-    conda config --system --set auto_update_conda false && \
-    conda config --system --set show_channel_urls true && \
-    conda update --all && \
-    conda update conda && \
-    conda install python==3.6.7 && \
-    conda clean -tipsy
-
 # NOTE: Beyond this point be careful of breaking out
 # or otherwise adding new layers with RUN, chown, etc.
 # The image size can grow significantly.
 
 # Install base python3 packages
-RUN pip install --upgrade pip==19.0.1 && \
+RUN pip install --upgrade pip==19.2.3 && \
     pip --no-cache-dir install \
-    # Tensorflow
-    ${TF_PACKAGE} \
-    # Jupyter Stuff
-    jupyter \
     jupyter-console==6.0.0 \
     jupyterhub \
     jupyterlab \
     xgboost \
     git+https://github.com/kubeflow/fairing@a9bb9d5cc1c9f1d75efa31198ddbdccfe176b7bf \
     # Kubeflow pipeline SDK
-    ${PIPELINE_SDK_PACKAGE} \
-    # Cleanup
-    && conda clean -tipsy
+    ${PIPELINE_SDK_PACKAGE} 
 
 # NB: the COPY chown can't expand a bash variable for NB_USER
-COPY --chown=jovyan:users requirements.txt /tmp
-
-# Install python2 and ipython2 kernel for jupyter notebook
-# Install tf packages which only support py2
-COPY --chown=jovyan:users install.sh /tmp/
-RUN chmod a+rx /tmp/install.sh && \
-    /tmp/install.sh
-
-# Add basic config
-COPY --chown=jovyan:users  jupyter_notebook_config.py /tmp
-
-# Wipe $HOME for PVC detection later
-WORKDIR $HOME
-RUN rm -fr $(ls -A $HOME)
-
-# Copy over init scripts
-COPY --chown=jovyan:users  pvc-check.sh /usr/local/bin/
-RUN chmod a+rx /usr/local/bin/*
+COPY --chown=${NB_USER}:users requirements.txt /tmp
 
-RUN docker-credential-gcr configure-docker && chown jovyan:users $HOME/.docker/config.json
+RUN docker-credential-gcr configure-docker && chown ${NB_USER}:users $HOME/.docker/config.json
 
 # Configure container startup
 EXPOSE 8888
-USER jovyan
+USER ${NB_USER}
```

Comment:
Do you need to declare this an Arg at the top of the dockerfile?

Reply:
Created an `ARG` for `NB_USER`.